File: src/main/java/io/github/robertomike/baradum/filters/NotEmptyFilter.java
package io.github.robertomike.baradum.filters;

import io.github.robertomike.baradum.requests.BasicRequest;
import io.github.robertomike.hefesto.actions.wheres.Where;
import io.github.robertomike.hefesto.builders.Hefesto;
import io.github.robertomike.hefesto.enums.Operator;

public class NotEmptyFilter extends Filter {
    public NotEmptyFilter(String field, String internalName) {
        super(field, internalName);
    }

    public NotEmptyFilter(String field) {
        super(field, field);
    }

    @Override
    public void filterByParam(Hefesto<?> query, BasicRequest<?> request) {
        if (request.findByName(field) != null) {
            filterByParam(query, "");
        }
    }

    @Override
    public void filterByParam(Hefesto<?> query, String value) {
        query.where(
                new Where(internalName, Operator.IS_NOT_NULL, null),
                new Where(internalName, Operator.DIFF, "")
        );
    }
}

File: src/main/java/io/github/robertomike/baradum/filters/NotEmptyFilter.kt
package io.github.robertomike.baradum.filters

import io.github.robertomike.baradum.requests.BasicRequest
import io.github.robertomike.hefesto.actions.wheres.Where
import io.github.robertomike.hefesto.builders.Hefesto
import io.github.robertomike.hefesto.enums.Operator

class NotEmptyFilter @JvmOverloads constructor(param: String, internalName: String = param) : Filter<Any>(param, internalName) {
    override fun filterByParam(query: Hefesto<*>, request: BasicRequest<*>) {
        if (request.findByName(param) != null) {
            filterByParam(query, "")
        }
    }

    override fun filterByParam(query: Hefesto<*>, value: String) {
        query.where(
            Where(internalName, Operator.IS_NOT_NULL, null),
            Where(internalName, Operator.DIFF, "")
        )
    }
}
